diff --git a/Telegram/Telegram.pro b/Telegram/Telegram.pro
index 56057e9..c30dfcc 100644
--- a/Telegram/Telegram.pro
+++ b/Telegram/Telegram.pro
@@ -511,11 +511,9 @@ PKGCONFIG += \
 
 LIBS += -ldl
 
-LIBS += $${QT_TDESKTOP_PATH}/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.a \
-        $${QT_TDESKTOP_PATH}/plugins/platforminputcontexts/libibusplatforminputcontextplugin.a \
-        $${QT_TDESKTOP_PATH}/plugins/platforminputcontexts/libfcitxplatforminputcontextplugin.a
-
-LIBS += ./../../../Libraries/breakpad/src/client/linux/libbreakpad_client.a
+LIBS += /usr/lib/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.a \
+        /usr/lib/plugins/platforminputcontexts/libibusplatforminputcontextplugin.a \
+        /usr/lib/plugins/platforminputcontexts/libfcitxplatforminputcontextplugin.a
 
 RESOURCES += \
 	./Resources/telegram.qrc \
