diff --git a/lib/xdg/telegramdesktop.desktop b/lib/xdg/telegramdesktop.desktop
index 0a80695..c165a74 100644
--- a/lib/xdg/telegramdesktop.desktop
+++ b/lib/xdg/telegramdesktop.desktop
@@ -2,7 +2,7 @@
 Version=1.0
 Name=Telegram Desktop
 Comment=Official desktop version of Telegram messaging app
-Exec=telegram-desktop -- %u
+Exec=telegram-desktop -- %u -platform flatpak
 Icon=telegram
 Terminal=false
 StartupWMClass=TelegramDesktop

